zerver/actions/user_groups.py

```diff
@@ -442,18 +443,32 @@ def do_change_user_group_permission_setting(
     old_value = getattr(user_group, setting_name)
     setattr(user_group, setting_name, setting_value_group)
     user_group.save()
+
+    audit_log_old_value = get_group_setting_value_for_api(old_value)
+    audit_log_new_value = get_group_setting_value_for_api(setting_value_group)
+
+    if not hasattr(old_value, "named_user_group") and hasattr(
+        setting_value_group, "named_user_group"
+    ):
+        # We delete the UserGroup which the setting was set to
+        # previously, if it does not have any linked NamedUserGroup
+        # object.
+        old_value.delete()
```

Comment:
Let's rework this comment to explain __why__ we're doing this; this comment kinda repeats what the code says.

Reply:
Updated the comment.